lib/workers/global/ssh_socket.ts

```diff
@@ -0,0 +1,47 @@
+import type { HostRule } from '../../types';
+import * as cp from 'child_process';
+import * as net from 'net';
+import Agent from 'ssh-agent-js/client/index';
+import { join } from 'upath';
+import { logger } from '../../logger';
+
+export class SshSocket {
+  agent_process: cp.ChildProcess | undefined;
+  agent: Agent | undefined;
+
+  async start(cacheDir: string) {
+    if (!process.env.SSH_AUTH_SOCK || process.env.SSH_AUTH_SOCK.length == 0) {
```

Comment:
```suggestion
    if (!is.nonEmptyOrWhitespaceString(process.env.SSH_AUTH_SOCK)) {
```
also invert logic to return early, so we have less nesting.

Reply:
No can do. We cannot switch to early return.

You see, if the env variable is set, we don't spawn the `ssh-agent`. But we still have to initialize the `Agent` class. Early return would mean that the agent is uninitialized and ssh would be unavailable.